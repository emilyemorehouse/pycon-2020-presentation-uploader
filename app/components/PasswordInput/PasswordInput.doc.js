import { describe, PropTypes } from 'react-desc'

export const doc = PasswordInput => {
  const DocumentedPasswordInput = describe(PasswordInput)
    .description('A PasswordInput')
    .usage(
      `import { PasswordInput } from 'components/PasswordInput';
<PasswordInput />`,
    )

  DocumentedPasswordInput.propTypes = {
    value: PropTypes.string.description('Value of the input').defaultValue(''),
    onChnage: PropTypes.func.description('Function to change the value').defaultValue(() => {}),
  }

  return DocumentedPasswordInput
}
