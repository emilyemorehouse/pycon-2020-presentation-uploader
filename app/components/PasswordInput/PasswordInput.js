import React from 'react'
import { View, Hide } from 'grommet-icons'

// Components
import { Box } from '../Box'
import { Button } from '../Button'
import { TextInput } from '../TextInput'

/**
 *
 * PasswordInput
 *
 * Displays a textinput with masked value for password
 *
 * - When View Icon is clicked, it reveals the password
 * - When HIde Icon is clicked, it masks the password
 *
 */
// eslint-disable-next-line react/prop-types
const PasswordInput = ({ value, onChange, type, ...rest }) => {
  if (type) {
    // eslint-disable-next-line no-console
    console.warn('type prop does not have an effect on PasswordInput!')
  }

  const [reveal, setReveal] = React.useState(false)
  return (
    <Box direction="row" align="center" height="44px">
      <TextInput
        onChange={e => onChange(e)}
        type={reveal ? 'text' : 'password'}
        value={value}
        {...rest}
      />

      <Button
        icon={
          reveal ? (
            <View size="large" style={{ width: 30 }} color="light-5" />
          ) : (
            <Hide size="xlarge" style={{ width: 30 }} color="light-5" />
          )
        }
        onClick={() => setReveal(!reveal)}
      />
    </Box>
  )
}

export default PasswordInput
