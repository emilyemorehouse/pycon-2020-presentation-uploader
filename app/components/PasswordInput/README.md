<!-- ! This is a generated file. To make changes, edit <Component>.doc.js ! -->
## PasswordInput
A PasswordInput

## Usage

```javascript
import { PasswordInput } from 'components/PasswordInput';
<PasswordInput />
```

## Properties

**value**

Value of the input

```
string
```

**onChnage**

Function to change the value Defaults to `function () {}`.

```
function
```
  