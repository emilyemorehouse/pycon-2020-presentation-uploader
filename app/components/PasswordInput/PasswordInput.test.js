import React from 'react'
import { render } from '@testing-library/react'
import { IntlProvider } from 'react-intl'

import PasswordInput from './PasswordInput'
import { DEFAULT_LOCALE } from '../../i18n'

const renderComponent = (props = {}) => render(<PasswordInput {...props} />)

test('it renders and matches snapshot', () => {
  const { container } = renderComponent()
  expect(container).toMatchSnapshot()
})

test('It does not log to the console', () => {
  const spy = jest.spyOn(global.console, 'log')
  render(
    <IntlProvider locale={DEFAULT_LOCALE}>
      <PasswordInput />
    </IntlProvider>,
  )
  expect(spy).not.toHaveBeenCalled()
})
