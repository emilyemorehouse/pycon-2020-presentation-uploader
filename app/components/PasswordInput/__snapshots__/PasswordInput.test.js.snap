// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`it renders and matches snapshot 1`] = `
<div>
  <div
    class="StyledBox-sc-13pk1d4-0 emmXTg"
  >
    <div
      class="StyledTextInput__StyledTextInputContainer-sc-1x30a0s-1 ksQrmr"
    >
      <input
        autocomplete="off"
        class="StyledTextInput-sc-1x30a0s-0 huJELu"
        type="password"
        value=""
      />
    </div>
    <button
      class="StyledButton-sc-323bzc-0 jvkqEN Button__StyledGrommetButton-sc-5cxguj-0 dKGxUf"
      type="button"
    >
      <svg
        aria-label="Hide"
        class="StyledIcon-ofa7kd-0 dhtznn"
        style="width: 30px;"
        viewBox="0 0 24 24"
      >
        <path
          d="M12,17 C9.27272727,17 6,14.2222222 6,12 C6,9.77777778 9.27272727,7 12,7 C14.7272727,7 18,9.77777778 18,12 C18,14.2222222 14.7272727,17 12,17 Z M11,12 C11,12.55225 11.44775,13 12,13 C12.55225,13 13,12.55225 13,12 C13,11.44775 12.55225,11 12,11 C11.44775,11 11,11.44775 11,12 Z M20,5 L4,19"
          fill="none"
          stroke="#000"
          stroke-width="2"
        />
      </svg>
    </button>
  </div>
</div>
`;
